returns/contrib/pytest/plugin.py

```diff
@@ -166,65 +118,107 @@ def _trace_function(
-        """We need this method so coverage will work correctly."""
-        from returns.context import (
-            RequiresContextFutureResult,
-            RequiresContextIOResult,
-            RequiresContextResult,
-        )
-        from returns.future import FutureResult
-        from returns.io import IOFailure, IOSuccess
-        from returns.result import Failure, Success
-
-        return (
-            Success,
-            Failure,
-            IOSuccess,
-            IOFailure,
-            RequiresContextResult,
-            RequiresContextIOResult,
-            RequiresContextFutureResult,
-            FutureResult,
-        )
+class _DesiredFunctionFound(BaseException):  # noqa: WPS418
+    """Exception to raise when expected function is found."""
 
-    @classmethod
-    def error_handler(cls, original):
-        if inspect.iscoroutinefunction(original):
-            async def factory(self, *args, **kwargs):
-                original_result = await original(self, *args, **kwargs)
-                _ERRORS_HANDLED[id(original_result)] = original_result
-                return original_result
-        else:
-            def factory(self, *args, **kwargs):
-                original_result = original(self, *args, **kwargs)
-                _ERRORS_HANDLED[id(original_result)] = original_result
-                return original_result
-        return wraps(original)(factory)
-
-    @classmethod
-    def copy_handler(cls, original):
-        if inspect.iscoroutinefunction(original):
-            async def factory(self, *args, **kwargs):
-                original_result = await original(self, *args, **kwargs)
-                if id(self) in _ERRORS_HANDLED:
-                    _ERRORS_HANDLED[id(original_result)] = original_result
-                return original_result
-        else:
-            def factory(self, *args, **kwargs):
-                original_result = original(self, *args, **kwargs)
-                if id(self) in _ERRORS_HANDLED:
-                    _ERRORS_HANDLED[id(original_result)] = original_result
-                return original_result
-        return wraps(original)(factory)
 
+def pytest_configure(config) -> None:
+    """
+    Hook to be executed on import.
 
-class _DesiredFunctionFound(BaseException):  # noqa: WPS418
-    """Exception to raise when expected function is found."""
+    We use it define custom markers.
+    """
+    config.addinivalue_line(
+        'markers',
+        (
+            'returns_lawful: all tests under `check_all_laws` ' +
+            'is marked this way, ' +
+            'use `-m "not returns_lawful"` to skip them.'
+        ),
+    )
+
+
+@pytest.fixture()
+def returns() -> Generator[ReturnsAsserts, None, None]:
+    """Returns class with helpers assertions to check containers."""
+    with _spy_error_handling() as errors_handled:
+        yield ReturnsAsserts(errors_handled)
+
+
+@contextmanager
+def _spy_error_handling() -> Generator[_ErrorsHandled, None, None]:
```

Comment:
Looks like it is just `Iterator[_ErrorsHandled]`

Reply:
Indeed -- I assumed `contextmanager` would complain -- let's test that hypothesis 🧑‍🔬 ...